src/site/nhentai.tsx

```diff
@@ -21,6 +21,11 @@ type Images = {
 };
 declare const _gallery: { num_pages: number; media_id: string; images: Images };
 
+function imageCdn() {
+  const cdn = unsafeWindow._n_app.options.image_cdn_urls as string[]
+  return cdn[Math.floor(Math.random() * cdn.length)];
```

Comment:
Ideally use PRNG generator seeded with gallery id to allow better client side request caching.